docs/content/wiring.md

```diff
@@ -0,0 +1,321 @@
+        "defaultValue": "wordpress",
+        "required": false,
+        "metadata": {},
+        "destination": {
+            "path": "",
+            "env": "DATABASE_NAME"
+        }
+    }
+}
+```
+
+## Wiring Parameters
+
+Once a parameter has been declared in the `porter.yaml`, Porter provides a simple mechanism for specifying how the parameter value should be wired into the mixin. To reference a parameter in the bundle, you use a notation of the form `source: bundle.parameters.PARAM_NAME`. This can be used anywhere within step definition. For example, to use the `database_name` parameter defined above in the `set` block of the helm mixin:
+
+```yaml
+install:
+- description: "Install MySQL"
+  helm:
+    name: porter-ci-mysql
+    chart: stable/mysql
+    version: 0.10.2
+    replace: true
+    set:
+      mysqlDatabase:
+        source: bundle.parameters.database-name
+      mysqlUser: "root"
+```
+
+Or to provide a parameter to the `command` attribute of the exec mixin:
+
+```yaml
+parameters:
+  - name: command
+    type: string
+    default: "echo Hello World"
+
+install:
+  - description: "Install Hello World"
+    exec:
+      command:
+        source:  bundle.parameters.command
+```
+
+This syntax is used in dictionary, as above, or in an list:
+
+```yaml
+parameters:
+  - name: command
+    type: string
+    default: "echo Hello World"
+
+install:
+  - description: "Install Hello World"
+    exec:
+      command: bash
+      arguments:
+        - -c
+        - source:  bundle.parameters.command
+```
+
+When the bundle is executed, the Porter runtime will locate the parameter definition in the `porter.yaml` to determine where the parameter value has been stored. The Porter runtime will then rewrite the YAML block before it is passed to the mixin. For example, given the YAML example above, the exec mixin will actually get the following YAML:
+
+```yaml
+description: "Install Hello World"
+exec:
+  command: bash
+  arguments:
+    - -c
+    - echo Hello World
+```
+
+## Credentials
+
+Credentials are defined in the `porter.yaml` with a YAML block of one more credential definitions. You can declare that a credential should be placed in a path within the invocation image or into an environment variable.
+
+To declare a file injection:
+
+```yaml
+credentials:
+- name: kubeconfig
+  path: /root/.kube/config
+```
+
+To declare an environment variable injection:
+
+```yaml
+credentials:
+- name: SUBSCRIPTION_ID
+  env: AZURE_SUBSCRIPTION_ID
+```
+
+## Wiring Credentials
+
+The same mechanism for declaring how to use a parameter can be used for credentials. To declare a credential usage, references are defined with the following syntax: `source: bundle.credentials.CREDENTIAL_NAME`.
+
+When the bundle is executed, the Porter runtime will locate the parameter definition in the `porter.yaml` to determine where the parameter value has been stored. The Porter runtime will then rewrite the YAML block before it is passed to the mixin.
```

Comment:
It feels like this is more information that is relevant to who is reading it (people who want to know how to wrire up a bundle). Maybe it belongs more in the runtime architecture?

Just worried that it's a bit overwhelming in addition to everything else. But it's okay to override me if you have strong feels.

Reply:
Removed, as we fill out that other doc we can incorporate how things are handled at runtime.